sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -171,6 +171,29 @@ trait FileFormat {
   def supportFieldName(name: String): Boolean = true
 }
 
+object FileFormat {
+
+  val FILE_PATH = "file_path"
+
+  val FILE_NAME = "file_name"
+
+  val FILE_SIZE = "file_size"
+
+  val FILE_MODIFICATION_TIME = "file_modification_time"
+
+  val METADATA_NAME = "_metadata"
+
+  // supported metadata struct fields for hadoop fs relation
+  val METADATA_STRUCT: StructType = new StructType()
+    .add(StructField(FILE_PATH, StringType))
+    .add(StructField(FILE_NAME, StringType))
+    .add(StructField(FILE_SIZE, LongType))
+    .add(StructField(FILE_MODIFICATION_TIME, LongType))
```

Comment:
should this be TimestampType?

Reply:
I think it's more like a design choice? I think both are fine, I don't have a strong opinion on it.
`long` matches what the file modification tells you directly
`timestamp` is more readable
WDYT?

Reply:
I think this one is an easy decision. Timestamp type is much better as people can do `WHERE _metadata.modificationTime < TIMESTAMP'2020-12-12 12:12:12'` or other datetime operations. And `df.show` can also display the value in a more user-readable format.

Reply:
got it, it makes sense! addressed.